http.hh

```diff
@@ -0,0 +1,19 @@
+#include <string>
+#include <unordered_map>
+
+enum http_type { get, post };
+
+struct http_request {
+  http_type type;
+};
+
+struct http_get : http_request {
+  std::string url;
+  std::unordered_map<std::string, std::string> args;
+};
+
+struct http_post : http_request {
+  std::string url;
+  int fd;
```

Comment:
what is fd? can we give it a better name?

Reply:
its the file descriptor where we will store the binary data.

Reply:
ill change it to file_descriptor

Reply:
wouldn't that be generated on the server side? we might not need this here and can just use the `http_request` struct for both get and post

Reply:
i was thinking that the serverside parses the requests into these data structures and passes them to the handlers.

Reply:
then we can have the lambdas that take in the http request as a parameter and returns some http response.